src/Network/Network/NetworkWatcher/ConnectionMonitor/SetAzureNetworkWatcherConnectionMonitorCommand.cs

```diff
@@ -79,178 +102,404 @@ public class SetAzureNetworkWatcherConnectionMonitorCommand : ConnectionMonitorB
         [Alias("ConnectionMonitorName")]
         [Parameter(
             Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
             HelpMessage = "The connection monitor name.",
-            ParameterSetName = "SetByName")]
+            ParameterSetName = "SetByResource")]
         [Parameter(
             Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
             HelpMessage = "The connection monitor name.",
-            ParameterSetName = "SetByResource")]
+            ParameterSetName = "SetByName")]
         [Parameter(
             Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
             HelpMessage = "The connection monitor name.",
             ParameterSetName = "SetByLocation")]
-        [ResourceNameCompleter("Microsoft.Network/networkWatchers/connectionMonitors", "ResourceGroupName", "NetworkWatcherName")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByResourceV2")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByNameV2")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor name.",
+            ParameterSetName = "SetByLocationV2")]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
+        [Parameter(
+              Mandatory = true,
+              HelpMessage = "The ID of the resource used as the source by connection monitor.",
+              ParameterSetName = "SetByResource")]
+        [Parameter(
+              Mandatory = true,
+              HelpMessage = "The ID of the resource used as the source by connection monitor.",
+              ParameterSetName = "SetByName")]
+        [Parameter(
+              Mandatory = true,
+              HelpMessage = "The ID of the resource used as the source by connection monitor.",
+              ParameterSetName = "SetByLocation")]
         [Parameter(
             Mandatory = true,
-            HelpMessage = "The ID of the connection monitor source.")]
+            HelpMessage = "The ID of the resource used as the source by connection monitor.",
+            ParameterSetName = "SetByResourceId")]
         [ValidateNotNullOrEmpty]
         public string SourceResourceId { get; set; }
 
         [Parameter(
-            Mandatory = false,
```

Comment:
This is a breaking change - the parameter cannot be made mandatory when it was not before.

Reply:
fixed